File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/read.cc
#include "read.h"
#include <iostream>
using namespace std;

void Read::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();
    cout << "? ";
    int a;
    cin >> a;

    pila.push(a);
    programa->inc();
}

string Read::mostrar(){
    return "read";
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Read.java
import java.util.Scanner;
import java.util.Stack;

public class Read extends Instruccion{

    private static final Scanner scanner = new Scanner(System.in);
    
    public void ejecutar(Programa programa){
        Stack<Integer> pila = programa.getPila();
        System.out.print("? ");
        int a = scanner.nextInt();
        pila.push(a);
        programa.inc();
    }

    public String mostrar(){
        return "read";
    }
}
